Improve ValueFormatter precision for small values

DRadmir opened the pull request:
Introduces a new formatterMiddle with 4 decimal places for values between 0.01 and 1 in ValueFormatter's auto style. Updates SwapSceneViewModel to use the .auto style and adjusts related tests to reflect the new formatting behavior.

![7bd0a7d9-4fea-4a66-8647-db21938d6edc](https://github.com/user-attachments/assets/f9a44ea2-aed4-4c32-ac0b-33634dc81f48)
![a291f5f7-7a38-4328-8552-493a517f15b4](https://github.com/user-attachments/assets/062157aa-7e56-489b-9dfd-154065e226e8)

DRadmir committed:
Improve ValueFormatter precision for small values

Introduces a new formatterMiddle with 4 decimal places for values between 0.01 and 1 in ValueFormatter's auto style. Updates SwapSceneViewModel to use the .auto style and adjusts related tests to reflect the new formatting behavior.

gemcoder21 committed:
Merge branch 'main' into value-formatter-impovements

gemcoder21 commented:
`#expect(formatter.string(11112344, decimals: 10) == "0.0011112344")` should be 
`#expect(formatter.string(11112344, decimals: 10) == "0.001111")`

gemcoder21 commented:
Left side look wront, right side is correct

<img width="320" alt="Screenshot 2025-07-08 at 5 27 11 AM" src="https://github.com/user-attachments/assets/4bfe50bc-4b7e-4fcf-ad06-5354ab3fd704" />

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into value-formatter-impovements

DRadmir committed:
Adjust ValueFormatter significant digits and tests

Updated ValueFormatter to use significant digits with a minimum of 2 and a maximum of 4, and enabled usesSignificantDigits. Modified the formatter selection logic and updated related unit tests to match the new formatting behavior.
